packages/flutter/lib/src/material/bottom_navigation_bar.dart

```diff
@@ -802,30 +817,35 @@ class _BottomNavigationBarState extends State<BottomNavigationBar> with TickerPr
     }
 
     ColorTween colorTween;
-    switch (widget.type) {
+    switch (_effectiveType) {
       case BottomNavigationBarType.fixed:
         colorTween = ColorTween(
-          begin: widget.unselectedItemColor ?? themeData.textTheme.caption.color,
-          end: widget.selectedItemColor ?? widget.fixedColor ?? themeColor,
+          begin: widget.unselectedItemColor
+            ?? bottomNavTheme.unselectedItemColor
+            ?? themeData.textTheme.caption.color,
+          end: widget.selectedItemColor
+            ?? bottomNavTheme.selectedItemColor
+            ?? widget.fixedColor
+            ?? themeColor,
         );
         break;
       case BottomNavigationBarType.shifting:
         colorTween = ColorTween(
-          begin: widget.unselectedItemColor ?? Colors.white,
-          end: widget.selectedItemColor ?? Colors.white,
+          begin: widget.unselectedItemColor ?? bottomNavTheme.unselectedItemColor ?? Colors.white,
```

Comment:
should Colors.white be colorScheme.surface?

Reply:
It's kind of ambiguous to be honest, because this is only for the `shifting` bottom navigation bars, which are legacy from older versions of material, before the `colorScheme` was introduced.

I also don't want to change it because that would be a breaking change to the default, and this change is non-breaking.